CONTRIBUTING.md

```diff
@@ -0,0 +1,66 @@
+# Contributing
+
+We welcome contributions to this project.
+
+## Repl
+
+For quick troubleshooting, there is a repl available by running `npm run repl`. This will start a Node.js repl with the
+`@pinecone-io/client` package preloaded. The `npm` command runs the file `utils/replInit.ts`.
+
+## Local testing
+
+To run all tests locally, excluding the unit tests, you will need to set your Pinecone API key to an environment
+variable (or hard-code it in a .env file in this repo).
+
+You can retrieve your API key from [app.pinecone.io](https://app.pinecone.io).
+
+```bash
+export PINECONE_API_KEY=your_api_key
+```
+
+To see the exact commands run by the aliases mentioned in this doc, see the `scripts` section in the `package.json`
+file.
+
+### Unit tests
+
+Simply run `npm run test:unit` to run all unit tests.
+
+### Integration tests
+
+Simply run `npm run test:integration-local:<runtime>` to run all integration tests, substituting `<runtime>` with
+either `node` or `edge` as you see fit.
+
+### End-to-end tests
+
+Simply run `npm run test:end-to-end -- <clone y/n> <path>` to run all end-to-end tests, passing in `y` or `n` for
+whether you want to clone the repo that contains the external app used for testing, and the `path` the repo should be
+cloned in, or is already cloned in, e.g.:
```

Comment:
I'm not necessarily clear on why we there's an option to clone or not clone the test app repo. What's the difference here?

I would think we would just clone it every time to make sure we're running against whatever is fresh there.

Reply:
Oh that's true, yes.... Okay I'll change!